projects/wvr-elements/src/lib/shared/wvr-base.component.ts

```diff
@@ -0,0 +1,110 @@
+  }
+
+  screenSizeChanged$: Observable<boolean>;
+
+  isMobileLayout = this.isMobileAgent;
+
+  protected readonly _animationService: WvrAnimationService;
+
+  protected readonly _domSanitizer: DomSanitizer;
+
+  protected readonly _eRef: ElementRef;
+
+  protected readonly _cdRef: ChangeDetectorRef;
+
+  @Output() protected readonly animationEventTrigger = new EventEmitter<Event>();
+
+  constructor(injector: Injector) {
+
+    this._animationService = injector.get(WvrAnimationService);
+    this._cdRef = injector.get(ChangeDetectorRef);
+    this._domSanitizer = injector.get(DomSanitizer);
+    this._eRef = injector.get(ElementRef);
+
+    this.screenSizeChanged$ = fromEvent(window, 'resize')
+      .pipe(debounceTime(50))
+      .pipe(map(this.checkScreenSize));
+  }
+
+  ngOnInit(): void {
+    const animationEvents = Object.keys(this._animationSettings);
+    if (animationEvents.length) {
+      if (this.animateId) {
+        this._animationService.registerAnimationReciever(this.animateId, this);
+      }
+      this.animationStateId = this._animationService.registerAnimationStates();
+      animationEvents.forEach(eventName => {
+        if (eventName !== 'animationTrigger') {
+          (this._eRef.nativeElement as HTMLElement).addEventListener(eventName, this.onEvent.bind(this));
+        }
+      });
+    }
+    this.screenSizeChanged$.subscribe(iml => {
+      this.isMobileLayout = iml;
+      this._cdRef.detectChanges();
+    });
+    this.isMobileLayout = this.checkScreenSize();
+  }
+
+  ngAfterContentInit(): void {
+    setTimeout(() => {
```

Comment:
Why is this timeout needed? Is there not lifecycle hook that providers the correct state to call initialize animation element?

Reply:
We tried all lifecycle hooks and were unable to find one in which both the ViewChild and the AnimationConfig were initialized. It was troubling, but this was the only way we discovered to ensure the presence of both.

Reply:
I was afraid of that.